presto-client/src/test/java/io/prestosql/client/auth/external/TestAuthenticationAssembler.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.client.auth.external;
+
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestAuthenticationAssembler
+{
+    @Test
+    public void testAssemblyForEmptyHeader()
+    {
+        assertThatThrownBy(() -> AuthenticationAssembler.toAuthentication(""))
+                .isInstanceOf(NullPointerException.class)
+                .hasMessageContaining("is null");
+    }
+
+    @Test
+    public void testAssemblyWithBothUrls()
+    {
+        ExternalAuthentication authentication = AuthenticationAssembler.toAuthentication("Bearer x_redirect_server=\"http://redirect.url\", x_token_server=\"http://token.url\"");
+
+        assertThat(authentication.getRedirectUrl()).hasValue("http://redirect.url");
+        assertThat(authentication.getTokenUrl()).isEqualTo("http://token.url");
+    }
+
+    @Test
+    public void testAssemblyWithAdditionalFields()
+    {
+        ExternalAuthentication authentication = AuthenticationAssembler.toAuthentication("Bearer type=\"token\", x_redirect_server=\"http://redirect.url\", x_token_server=\"http://token.url\", description=\"oauth challenge\"");
+
+        assertThat(authentication.getRedirectUrl()).hasValue("http://redirect.url");
+        assertThat(authentication.getTokenUrl()).isEqualTo("http://token.url");
+    }
+
+    @Test
+    public void testAssemblyWithOnlyTokenServerUrl()
+    {
+        ExternalAuthentication authentication = AuthenticationAssembler.toAuthentication("Bearer x_token_server=\"http://token.url\"");
+
+        assertThat(authentication.getRedirectUrl()).isEmpty();
+        assertThat(authentication.getTokenUrl()).isEqualTo("http://token.url");
+    }
```

Comment:
How about:
 -  `Bearer ma kota`
 - `Bearer x_redirect_server=x_redirect_server=x_redirect_server=kot`?
 - other crazy stuff

Reply:
it will all throw a NPE exception, but I could make it better.